alea/include/alps/alea/computed.hpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 1998-2017 ALPS Collaboration. See COPYRIGHT.TXT
+ * All rights reserved. Use is subject to license terms. See LICENSE.TXT
+ * For use in publications, see ACKNOWLEDGE.TXT
+ */
+#pragma once
+
+#include <cstddef>
+#include <cassert>
+#include <stdexcept>
+#include <complex>
+
+#include <vector>
+#include <Eigen/Dense>
+
+#include <alps/alea/core.hpp>
+#include <alps/alea/util.hpp>
+
+namespace alps { namespace alea {
+
+// Forward declarations
+
+struct size_mismatch;
+struct unsupported_operation;
+template <typename T> class sink;
+template <typename T> struct computed;
+template <typename T, typename E> class computed_adapter;
+
+// Actual declarations
+
+/** Promote type to computed result */
+template <typename T, typename Estimator>
+class computed_adapter;
+
+template <typename T>
+class computed_adapter<T, T>
+    : public computed<T>
+{
+public:
+    typedef T value_type;
+
+public:
+    computed_adapter(T in) : in_(in) { }
+
+    size_t size() const { return 1; }
+
+    void add_to(sink<T> out) const
+    {
+        if (out.size() != 1)
+            throw size_mismatch();
+        out.data()[0] += in_;
+    }
+
+    ~computed_adapter() { }
+
+private:
+    T in_;
+};
+
+template <typename T>
+class computed_adapter< T, std::vector<T> >
+    : public computed<T>
+{
+public:
+    typedef T value_type;
+
+public:
+    computed_adapter(const std::vector<T> &in) : in_(in) { }
+
+    size_t size() const { return in_.size(); }
+
+    void add_to(sink<T> out) const
+    {
+        if (out.size() != in_.size())
+            throw size_mismatch();
+        for (size_t i = 0; i != in_.size(); ++i)
+            out.data()[i] += in_[i];
+    }
+
+    ~computed_adapter() { }
+
+private:
+    const std::vector<T> &in_;
+};
+
+
+template <typename T>
+class computed_adapter< T, column<T> >
+    : public computed<T>
+{
+public:
+    typedef T value_type;
+
+public:
+    computed_adapter(const column<T> &in) : in_(in) { }
+
+    size_t size() const { return in_.size(); }
+
+    void add_to(sink<T> out) const
+    {
+        if (out.size() != (size_t)in_.rows())
+            throw size_mismatch();
+        for (size_t i = 0; i != (size_t)in_.rows(); ++i)
```

Comment:
Good question. Originally, because I did not want to have a hard Eigen dependency in the interface.  Now, since its everywhere anyway, I don't mind removing the `sink<T>` class altogether and use `Eigen::Map<Eigen::Matrix<T, 1, Dynamic> >` instead. It is a little bit a question of taste, since `sink<T>` is quite a bit more lightweight . . .